app/javascript/packages/document-capture/higher-order/with-props.tsx

```diff
@@ -0,0 +1,13 @@
+// withProps returns a function that accepts a JSX component and binds
+// some new props to that JSX component, then returns it.
+function withProps(boundProps) {
+  function bindProps(Component) {
+    function ComponentWithBoundProps(props) {
+      return <Component {...boundProps} {...props} />;
+    }
+    return ComponentWithBoundProps;
```

Comment:
what if we copied over the name from `Component`? Otherwise these will all have the same name
```suggestion
    ComponentWithBoundProps.displayName = `${Component.displayName}.WithProps`
    return ComponentWithBoundProps;
```

Reply:
This is old documentation, but there's also some React conventions we could adopt here:

https://legacy.reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging

Reply:
Added some better naming and a test for it ([commit](https://github.com/18F/identity-idp/pull/10527/commits/b8bcd9890eba8a0c24f5abc77f5d6695cb8ae9a9)).

Reply:
I'm kinda confused how that works 🤔  Does React internally create the wrapping display name with `displayName`?

Either way, if it works it works! 😄

Reply:
Seems like they decided that the code in the docs you [linked](https://legacy.reactjs.org/docs/higher-order-components.html#convention-wrap-the-display-name-for-easy-debugging) should just be part of their codebase [here](https://github.com/facebook/react/blob/4508873393058e86bed308b56e49ec883ece59d1/packages/shared/getComponentNameFromType.js#L45).

Reply:
Neat! Does that mean we could just do something like... ? Or would the specs fail?

```tsx
ComponentWithBoundProps.displayName = Component.displayName;
```

Reply:
Oh, nice! That works perfectly ([commit](https://github.com/18F/identity-idp/pull/10527/commits/66b11d6706a207a4581b070cd57e0738c4e72168)).